README.md

```diff
@@ -0,0 +1,101 @@
+# Fastify Secrets HashiCorp
+
+![CI](https://github.com/nearform/fastify-secrets-hashicorp/workflows/CI/badge.svg)
+
+Fastify secrets plugin for [HashiCorp Vault](https://www.vaultproject.io).
+
+## Installation
+
+```sh
+npm install --save fastify-secrets-hashicorp
+```
+
+## Usage
+
+### Add plugin to your fastify instance
+
+```js
+fastify.register(FastifySecretsHashiCorp, {
+  secrets: {
+    dbPassword: 'secret-name'
+  }
+})
+```
+
+### Access your secrets
+
+```js
+await fastify.ready()
+
+console.log(fastify.secrets.dbPassword) // content of 'secret-name'
+```
+
+### Plugin options
+
+The plugin can be initialised with clientOptions as follows:
+
+```js
+fastify.register(FastifySecretsHashiCorp, {
+  secrets: {
+    dbPassword: 'secret-name'
+  },
+  clientOptions: {
+  }
+})
+```
+
+Following the structure
+
+```js
+clientOptions: {
+    vaultOptions: {
+      token: '', 
+      endpoint: ''
+    },
+    mountPoint, ''
+  }
+```
+
+#### clientOptions.mountPoint
+
+The [path to the secrets engine](https://www.vaultproject.io/docs/secrets#secrets-engines-lifecycle). Defaults to 'secret'.
+
+#### clientOptions.vaultOptions
+
+Initialisation options that are sent to [node-vault](https://github.com/kr1sp1n/node-vault), typed as [VaultOptions](https://github.com/kr1sp1n/node-vault/blob/master/index.d.ts#L115).
```

Comment:
use a permalink otherwise if this file changes in master this link may point somewhere else. either that, or simply point to the file without a specific line number